compiler/optimizer/optimizer.go

```diff
@@ -110,6 +110,16 @@ func (o *Optimizer) OptimizeScan() error {
 	}
 	for k := range from.Trunks {
 		pushDown(&from.Trunks[k])
+		// Check to see if we can add a range pruner when the pool-key is used
+		// in a normal filtering operation.
+		trunk := &from.Trunks[k]
+		if layout, ok := o.layouts[trunk.Source]; ok {
+			if pushdown, ok := trunk.Pushdown.(*dag.Filter); ok {
+				if p := newRangePruner(pushdown.Expr, layout.Primary(), layout.Order); p != nil {
+					trunk.KeyPruner = p
+				}
```

Comment:
I might be missing something, but it trunk.KeyPruner looks to me as though it's always nil here, so you could simplify a little.
```suggestion
				trunk.KeyPruner = newRangePruner(pushdown.Expr, layout.Primary(), layout.Order)
```

Reply:
This avoids the dreaded non-nil interface that points to a concrete nil pointer.  I could change rangePruner to return the interface instead of the struct pointer.  Which pattern do you prefer?

Reply:
Ah. Stick with what you've got.